File: NasleGhalam.ViewModels/Province/ProvinceViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace NasleGhalam.ViewModels.Province
{
    public class ProvinceViewModel
    {
        public int Id { get; set; }


        [Required(ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "Required")]
        [MaxLength(50, ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "MaxLen")]
        [Display(Name = "نام")]
        public string Name { get; set; }


        [Required(ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "Required")]
        [MaxLength(5, ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "MaxLen")]
        [Display(Name = "پیش شماره")]
        public string Code { get; set; }
    }
}

File: NasleGhalam.ClientApp/src/viewModels/provinceViewModel.js
import { required, displayName, maxLength } from 'plugins/vuelidate';

export default {
  provinceObj: {
    Name: {
      displayName: displayName('نام'),
      maxLength: maxLength(50),
      required
    },
    Code: {
      displayName: displayName('کد'),
      maxLength: maxLength(5),
      required
    }
  }
};
